### PR TITLE
[WIP] Removing Asset

### PR DESCRIPTION
**Pull request by majecty**

### CODE REVIEW

File: state/src/item/metadata.rs [hunk 1]
```diff
@@ -110,23 +77,24 @@ impl Metadata {
...
     }
 }
 
 impl CacheableItem for Metadata {
     type Address = MetadataAddress;
 
+    // FIXME: I'm not sure of it.
     fn is_null(&self) -> bool {
-        self.number_of_shards == 0
+        self.hashes.is_empty() && self.seq == 0
```
[the remark this answers is not in the document]
Comment (majecty) [reply state/src/item/metadata.rs]: @sgkim126 IMO this code works find. What do you think about it?

**Comment by majecty** [comment]
There are two failed e2e tests.

```
  1) Memory pool memory limit test
       To others
         More than limit:

      AssertionError: expected 5 to equal 0
      + expected - actual

      -5
      +0

      at Context.<anonymous> (src/e2e.long/mempool.test.ts:206:64)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)

  2) sync 2 nodes
       2 nodes
         A-B diverged with the same transaction
           Both transaction success
             "before each" hook for "It should be synced when A-B connected":

      AssertionError: expected { Object (value) } to not deeply equal { Object (value) }
      + expected - actual


      at Context.<anonymous> (src/e2e.long/sync2.test.ts:170:72)
      at processTicksAndRejections (internal/process/task_queues.js:85:5)
```

File: core/src/miner/miner.rs [hunk 2]
```diff
@@ -534,7 +482,7 @@ impl Miner {
         let tx_total = transactions.len();
         let mut invalid_tx_users = HashSet::new();
 
-        let immune_users = self.immune_users.read();
+        let _immune_users = self.immune_users.read();
```
Comment (Remagpie) [inline core/src/miner/miner.rs]: I think we can remove this.

File: core/src/scheme/scheme.rs [hunk 3]
```diff
@@ -133,6 +134,21 @@ impl Scheme {
...
+            let address = MetadataAddress::new();
+
+            let r = t.insert(&*address, &global_metadata.rlp_bytes());
+            debug_assert_eq!(Ok(None), r);
+            r?;
+        }
+
+        Ok((db, root))
+    }
+
```
Comment (Remagpie) [inline core/src/scheme/scheme.rs]: I suggest moving this change to a separated commit.
I think it's too important to be aggregated with removing tracket and vm.

File: types/src/errors/syntax_error.rs [hunk 4]
```diff
@@ -77,7 +62,6 @@ const ERROR_ID_TEXT_CONTENT_TOO_BIG: u8 = 24;
 const ERROR_ID_TOO_MANY_OUTPUTS: u8 = 26;
 const ERROR_ID_TX_IS_TOO_BIG: u8 = 27;
 const ERROR_ID_ZERO_QUANTITY: u8 = 28;
-const ERROR_ID_CANNOT_CHANGE_WCCC_ASSET_SCHEME: u8 = 29;
 const ERROR_ID_DISABLED_TRANSACTION: u8 = 30;
 const ERROR_ID_INVALID_SIGNER_OF_WRAP_CCC: u8 = 31;
```
Comment (Remagpie) [inline types/src/errors/syntax_error.rs]: We have some remaining error ids. Let's remove them.

**Comment by sgkim126** [comment]
It seems that this commit removes a shard itself not only assets. Is it right?
IMO, it's better to keep shards and shard manipulation transactions like change owners, create shards, etc.

File: state/src/item/metadata.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgkim126) [reply state/src/item/metadata.rs]: I think the code is right except we should remove hashes too to remove shards.
However, IMO, we've better keep shards.

[the remark this answers is not in the document]
Comment (sgkim126) [reply state/src/item/metadata.rs]: FYI, `is_null` should be `true` only when the value means that it does not exist in the trie.

**Comment by majecty** [comment]
@sgkim126 I removed shard because I removed all shard transactions. I tried to inform you in Gitter, before removing shards, but it seems not delivered to you. I'll care about communication more.

**Comment by majecty** [comment]
I'll restore Shard code back.

**Comment by sgkim126** [comment]
> @sgkim126 I removed shard because I removed all shard transactions. I tried to inform you in Gitter, before removing shards, but it seems not delivered to you. I'll care about communication more.

What I understand was you would remove all transactions that change the states of shards. I reread your message. It was my confusion.

**Comment by majecty** [comment]
I'm creating small PRs that only removes the asset related code.
So we don't need this PR anymore.

**Pull request closed without merging**